### DISCUSSION TITLE
Listening event of node selection v2

### DISCUSSION THREAD

**Discussion by code-snips** [newcat/baklavajs#396] [Q&A] [answered]
Is it possible to add an eventhandler listening to node selections in baklava v2 with vuejs3 and composition api?

Edit:
I still try to log nodeIds when they are selected. I have found some additional recources:

a similar question was asked in V1, but the selectedNodes is no longer available https://github.com/newcat/baklavajs/issues/160

I saw in the documentation that nodes now have a prop “selected”: https://v2.baklava.tech/visual-editor/customization.html

select (type: () => void) callback for node being selected.
how can I pass a function to the node?
I only call defineNode and then register it:
export default defineNode({

    type: NodeTypes.SOURCE,
    title: "Source",
    inputs: {
    },
    outputs: {
out1: () => new NodeInterface("rgb", new Mat()).use(setType, sourceType),
    },
});
baklava.editor.registerNodeType(Source, { category: "Core" });

I appreciate any suggestions. Keep up the great work!

**Comment by newcat** [comment, marked as answer]
The selected nodes are now part of the graph. You can watch the selected nodes:
```ts
const baklava = useBaklava();
watch(() => baklava.displayedGraph.selectedNodes, () => {
  // your logic
});
```

**Reply by code-snips** [reply]
thank you!

**Comment by code-snips** [comment]
its not possible for me to access  the .selectedNodes property.
the minimal example:
```
<template>
    <v-container>
        <div style="width: 100%; height: 80vh">
            <BaklavaEditor :view-model="baklava" />
        </div>
    </v-container>
</template>

<script setup lang="ts">
import { ref, onMounted, onUnmounted, watch } from "vue";
import { BaklavaEditor, useBaklava, Commands } from "@baklavajs/renderer-vue";

const baklava = useBaklava();
watch(() => baklava.displayedGraph.selectedNodes, () => {  // ERROR here: Property 'selectedNodes' does not exist on type 'Graph'.
    console.log("selectedNodes", baklava.displayedGraph.selectedNodes);
});
</script>
```
I installed the lastest version of the library `"baklavajs": "^2.4.3",`
I also tried to install the packages from the [tutorial](https://v2.baklava.tech/getting-started.html#installation) separatly, but the property is not recognized.
Currently, i am using a workaround using MutationObserver watching '--selected' tags, which is obviously not ideal.

**Reply by newcat** [reply]
Should now be fixed - the type overrides were not exported in the `renderer-vue` package, meaning that TypeScript didn't know about the property. And I wasn't able to reproduce it in the playground, since the overrides are included in there.

**Reply by code-snips** [reply]
I tested version 2.4.4. Works fine! Thanks for the patch.
